File: back/Context/Models/Ponto.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SGE.Context.Models
{
    public class Ponto : GenericModel
    {
        [Required]
        public DateTimeOffset Data { get; set; }

        [Required]
        [ForeignKey("UsuarioId")]
        public long UsuarioId { get; set; }
        public Usuario Usuario { get; set; }

    }
}

File: front/src/app/models/ponto.ts
import { GenericModel } from './genericModel';
import { Usuario } from './usuario';

export class Ponto extends GenericModel {
    data: Date;
    tarefa: string;

    usuarioId: number;
    usuario: Usuario;

    constructor(item?) {
        super(item);

        if (item) {
            this.data = new Date(item.data);
            this.tarefa = item.tarefa;
            this.usuarioId = item.usuarioId;

            if (item.usuario)
                this.usuario = new Usuario(item.usuario);
        }
    }
}
